src/Components/App/App.js

```diff
@@ -18,37 +18,52 @@ class App extends Component {
       bestStoryIDs: [],
       topStoryIDs: [],
       homePageStories: [],
+      test: ''
     };
   }
 
-  componentDidMount = () => {
-    fetchPromises("newstories")
+  componentDidMount = async () => {
+    await fetchPromises("newstories")
       .then((data) => this.setState({ newStoryIDs: data }))
       .then((newData) => this.getStories(this.state.newStoryIDs[0]))
+      .then(data => this.addTopic(data, 'Newest'))
       .catch((err) => console.error(err));
 
-    fetchPromises("beststories")
+    await fetchPromises("beststories")
       .then((data) => this.setState({ bestStoryIDs: data }))
       .then((newData) => this.getStories(this.state.bestStoryIDs[0]))
+      .then(data => this.addTopic(data, 'Best'))
       .catch((err) => console.error(err));
 
-    fetchPromises("topstories")
+    await fetchPromises("topstories")
       .then((data) => this.setState({ topStoryIDs: data }))
       .then((newData) => this.getStories(this.state.topStoryIDs[0]))
+      .then(data => this.addTopic(data, 'Top'))
       .catch((err) => console.error(err));
   };
 
   getStories = async (id) => {
     const story = await fetchStories(id);
     this.setState({ homePageStories: [...this.state.homePageStories, story] });
+    return story;
+  };
+
+  addTopic = (story, topic) => {
+    const matchingStory = this.state.homePageStories.find(specificStory => specificStory.id == story.id)
+    matchingStory['topic'] = topic;
+    const index = this.state.homePageStories.indexOf(matchingStory);
+    this.state.homePageStories.splice(index, 1);
+    this.setState({
+      homePageStories: [...this.state.homePageStories, matchingStory]
+    })
```

Comment:
Finds the story and adds the correct topic

Reply:
teamwork makes the dreamwork.